tests/std/tests/P3503R3_packaged_task_promise_with_allocator/test.cpp

```diff
@@ -0,0 +1,250 @@
+    template <class U>
+    constexpr global_counting_allocator(global_counting_allocator<U>) noexcept {}
+
+    T* allocate(size_t n) {
+        auto p = allocator<T>{}.allocate(n);
+        ++global_allocation_count;
+        return p;
+    }
+
+    void deallocate(T* p, size_t n) {
+        allocator<T>{}.deallocate(p, n);
+        ++global_deallocation_count;
+    }
+
+    template <class U>
+    friend constexpr bool operator==(global_counting_allocator, global_counting_allocator<U>) noexcept {
+        return true;
+    }
+#if !_HAS_CXX20
+    template <class U>
+    friend constexpr bool operator!=(global_counting_allocator, global_counting_allocator<U>) noexcept {
+        return false;
+    }
+#endif // !_HAS_CXX20
+};
+
+template <class T>
+void test_implicit_conversion(T) noexcept;
+
+template <class V, class T, class... Args>
+constexpr bool is_copy_list_initializable_impl = false;
+template <class T, class... Args>
+constexpr bool
+    is_copy_list_initializable_impl<decltype(test_implicit_conversion<T>({declval<Args>()...})), T, Args...> = true;
+
+template <class T, class... Args>
+constexpr bool is_copy_list_initializable = is_copy_list_initializable_impl<void, T, Args...>;
+
+template <class R>
+struct large_functor {
+    unsigned char dummy_[64]{};
+
+    template <class... Args>
+    R operator()(Args&&...) const {
+        return R{};
+    }
+};
+template <class R>
+struct large_functor<R&> {
+    unsigned char dummy_[64]{};
+
+    template <class... Args>
+    R& operator()(Args&&...) const {
+        static R r{};
+        return r;
+    }
+};
+template <>
+struct large_functor<void> {
+    unsigned char dummy_[64]{};
+
+    template <class... Args>
+    void operator()(Args&&...) const {}
+};
+
+STATIC_ASSERT(is_constructible_v<promise<int>>);
+STATIC_ASSERT(is_constructible_v<promise<int>, const allocator_arg_t&, const allocator<int>&>);
+STATIC_ASSERT(is_constructible_v<promise<int>, const allocator_arg_t&, const global_counting_allocator<int>&>);
+
+STATIC_ASSERT(is_copy_list_initializable<promise<int>>);
+STATIC_ASSERT(is_copy_list_initializable<promise<int>, const allocator_arg_t&, const allocator<int>&>);
+STATIC_ASSERT(is_copy_list_initializable<promise<int>, const allocator_arg_t&, const global_counting_allocator<int>&>);
+
+STATIC_ASSERT(is_constructible_v<packaged_task<int()>>);
+STATIC_ASSERT(
+    is_constructible_v<packaged_task<int()>, const allocator_arg_t&, const allocator<int>&, large_functor<int>>);
+STATIC_ASSERT(is_constructible_v<packaged_task<int()>, const allocator_arg_t&, const global_counting_allocator<int>&,
+    large_functor<int>>);
+
+STATIC_ASSERT(is_copy_list_initializable<packaged_task<int()>>);
+STATIC_ASSERT(!is_copy_list_initializable<packaged_task<int()>, const allocator_arg_t&, const allocator<int>&,
+    large_functor<int>>);
+STATIC_ASSERT(
+    !is_copy_list_initializable<packaged_task<int()>, const allocator_arg_t&, const global_counting_allocator<int>&>);

```

Comment:
This is missing the functor.

(How much caffeine did I need to find this? Over 300 mg! *I CAN SEE TIME*)